### PR TITLE
[Traceable FSDP2] Fix support for CUDA resize_storage_bytes_

### PR DESCRIPTION
**Pull request by yf225**
Currently if `x` is a CUDA tensor, calling `x.untyped_storage().resize_()` seems to always go into the `built without cuda` branch of `resize_storage_bytes_()` regardless of whether PyTorch is built with CUDA. I suspect this is because `inductor_ops.cpp` is only included in `libtorch_cpu.so` thus doesn't have the `USE_CUDA` information or ability to link to CUDA-related functions.

This PR moves `resize_storage_bytes_()` related custom op functions out of `inductor_ops.cpp` into its standalone file `resize_storage_bytes.cpp` to be included in `libtorch_python.so` instead. This mimics the setup for `StorageMethods.cpp`. This way, `resize_storage_bytes_()` can have access to the CUDA-related functions, which passes the CUDA unit test.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129157
* #129203
* __->__ #129215

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: test/inductor/test_distributed_patterns.py [hunk 1]
```diff
@@ -146,20 +146,23 @@ def fn(x, obj):
         self.assertEqual(x1.grad, x3.grad)
 
     @torch.no_grad()
+    @requires_gpu()
```
Comment (jansel) [inline test/inductor/test_distributed_patterns.py:149]: We should make this work on CPU and CUDA.

File: test/inductor/test_distributed_patterns.py [hunk 2]
```diff
@@ -146,20 +146,23 @@ def fn(x, obj):
...
-        self.assertEqual(x.untyped_storage().size(), 0)
+        for device in ("cpu", "cuda"):
+            x = torch.randn(10, device=device)
+            expected = torch.cos(torch.sin(x))
+            y = fn(x)
+            self.assertEqual(y, expected)
+            self.assertEqual(x.untyped_storage().size(), 0)
 
     @torch.no_grad()
+    @requires_gpu()
```
Comment (jansel) [inline test/inductor/test_distributed_patterns.py:165]: We should make this work on CPU and CUDA.

**Review by jansel — CHANGES_REQUESTED** [review]
Seems fine to me, but breaking CPU support is a problem.

File: test/inductor/test_distributed_patterns.py (hunk 2 above)
Comment (yf225) [reply test/inductor/test_distributed_patterns.py:166]: Updated to split into two tests: CPU and CUDA

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**